core/meterer/onchain_state.go

```diff
@@ -101,15 +157,27 @@ func (pcs *OnchainPaymentState) GetActiveReservationByAccount(ctx context.Contex
 	return res, nil
 }
 
+// GetActiveReservationByAccountOnChain returns on-chain reservation for the given account ID
+func (pcs *OnchainPaymentState) GetActiveReservationByAccountOnChain(ctx context.Context, accountID string) (core.ActiveReservation, error) {
+	blockNumber, err := pcs.tx.GetCurrentBlockNumber(ctx)
+	if err != nil {
+		return core.ActiveReservation{}, err
+	}
+	res, err := pcs.tx.GetActiveReservationByAccount(ctx, blockNumber, accountID)
+	if err != nil {
+		return core.ActiveReservation{}, errors.New("reservation account not found on-chain")
```

Comment:
nit: wrap the actual error when returning the error, i.e. `fmt.Errorf("reservation account not found on-chain: %w", err)`

Reply:
thanks, updated